pkgs/misc/tmux-plugins/tmux-which-key/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  mkTmuxPlugin,
+  fetchFromGitHub,
+  lib,
+  check-jsonschema,
+  python3,
+}:
+mkTmuxPlugin {
+  pluginName = "tmux-which-key";
+  rtpFilePath = "plugin.sh.tmux";
+  version = "0-unstable-2024-06-08";
+  buildInputs = [
+    check-jsonschema
+    (python3.withPackages (ps: with ps; [ pyyaml ]))
+  ];
+
+  postPatch = ''
+    sed -i -e 's|python3 |${python3}/bin/python3 |g' plugin.sh.tmux
+  '';
+
+  preInstall = ''
+    rm -rf plugin/pyyaml
+    ln -s ${python3.pkgs.pyyaml.src} plugin/pyyaml
```

Comment:
Given that we pass in a Python with the pyyaml package available above, we can skip settings this link, I would assume.

Reply:
The upstream package [actually has a copy of pyyaml linked in it](https://github.com/alexwforsythe/tmux-which-key/tree/main/plugin), and the build script that it uses to create its configurations also [imports from that linked library](https://github.com/alexwforsythe/tmux-which-key/blob/1f419775caf136a60aac8e3a269b51ad10b51eb6/plugin/build.py#L10C1-L10C28). I included that in case that imported library would cause issues with the package building its configuration file (also I was following [another example of a nixpkgs package](https://github.com/alexwforsythe/tmux-which-key/pull/9/files#diff-4e3bd328e4c998f53b19da7639df45b2db63601be4b4343534880adfbeaa5840) that the repo includes).

Reply:
Just confirmed my suspicion, if I took that entire line out then the configuration build script (used with home-manager or even manually calling the script) fails to execute because it imports the library (pyyaml.lib).